modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java

```diff
@@ -69,10 +74,43 @@ public Set<Integer> getLivenessCheckPortNumbers() {
     @Override
     public String getDriverClassName() {
         try {
-            Class.forName(DRIVER_CLASS_NAME);
-            return DRIVER_CLASS_NAME;
+            if (supportsNewDriver) {
+                Class.forName(DRIVER_CLASS_NAME);
+                return DRIVER_CLASS_NAME;
+            } else {
+                return LEGACY_DRIVER_CLASS_NAME;
+            }
         } catch (ClassNotFoundException e) {
-            return "com.clickhouse.jdbc.ClickHouseDriver";
+            return LEGACY_DRIVER_CLASS_NAME;
+        }
+    }
+
+    private static boolean isNewDriverSupported(DockerImageName dockerImageName) {
```

Comment:
I think you may want to use `ComparableVersion`. See https://github.com/testcontainers/testcontainers-java/blob/96af052165f46e1b7b573bc74181c5bbdeb5c725/modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java#L110

Reply:
@eddumelendez Oh, that's great, thanks! Will fix it as soon as possible.

Reply:
Replaced with `ComparableVersion`.